PerDiem/PerDiem/models/PerDiem.m

```diff
@@ -107,4 +107,13 @@ + (void)perDiemsForDate:(NSDate *)date
                         }];
 }
 
++ (BOOL) souldUsePendingTransaction: (Transaction*) transaction  day:(NSDate*) day {
+    NSDate *startOfDay = [[NSCalendar currentCalendar] startOfDayForDate:day];
+    NSDate *nextDay = [startOfDay dateByAddingDays:1];
+
+    return [transaction.future boolValue] && ([transaction.createdAt earlierDate:day] ||
```

Comment:
The first scenario can't happen, because we are only fetching transactions that belongs to the current period. 

I added the createdAt logic because i wanted to keep the perdiem factor as consistent as we can. Let say on Sunday you have a perDiem of $100 and you spend the $100, if next monday you add a future transaction and you perdiem goes down to $90 it will looks like on Sunday you spent more than you had to, when in reality you didn't. 

I don't have a strong opinion on this. Let me know your thoughts.